train.py

```diff
@@ -252,6 +252,11 @@ def loss_fn(pred, labels):
         for m in model_parts
     ]
 
+    # for ease of testing TP in lieu of FSDP
+    if job_config.training.tensor_parallel_degree == world_size:
+        for model in model_parts:
+            model.to(torch.bfloat16)
```

Comment:
May I ask what are we doing here?

Reply:
Hey @tianyu-l, this is to allow conveniently testing `tp_degree=8` in bfloat16 on a single host. When running with `tp_degree=8` on a single 8-GPU host, FSDP will not be used and the model will be in float32. I figured since we don't care about float32, we might as well move the model to bfloat16 when FSDP is not used.

I can remove this if we feel it is janky. This logic is not related to async-TP.

Reply:
oh I see... I thought we still care about fp32 in TP because bf16 is lossy when doing reductions. Maybe I'm lacking some context.

Even if we would offer precision adjustment for TP, I feel we should have an option for user to decide as in FSDP, like [here](https://github.com/pytorch/torchtitan/blob/main/torchtitan/config_manager.py#L310).

Reply:
Sounds good. I'll remove this for now and explore other options later